### DISCUSSION TITLE
Passing complex object as dependencies to components

### DISCUSSION THREAD

**Discussion by cosmincatalin** [DioxusLabs/dioxus#2305] [Q&A] [answered]
How would one go about using, say, an `Arc<Mutex<String>>` as argument to a component? As it is now, the compiler complains about `Arc` not implementing `PartialEq`.

**Comment by ealmloff** [comment, marked as answer]
You can implement PartialEq manually with an external props struct (not the `#[component]` macro). In that manual PartialEq implementation, you can skip diffing that field if you don't need to rerun the component if the field changes
